datasource/src/main/scala/quasar/blobstore/azure/AzureBlobstore.scala

```diff
@@ -124,12 +129,12 @@ class AzureBlobstore[F[_]: ConcurrentEffect: MonadResourceErr: RaiseThrowable](
   private def pathToOptions(path: ResourcePath): ListBlobsOptions =
     new ListBlobsOptions()
       .withMaxResults(Integer.valueOf(5000))
-      .withPrefix(pathToPrefix(path))
+      .withPrefix(normalizePrefix(pathToPrefix(path)))
 
   private def pathToPrefix(path: ResourcePath): String = {
     val names = ResourcePath.resourceNamesIso.get(path).map(_.value).toList
-    if (names.isEmpty) ""
-    else names.mkString("", "/", "/")
+    val s = names.mkString("", "/", "/")
+    if (s === "/") "" else s
```

Comment:
I guess it doesn't hurt, but normalization would deal with this, no?

Reply:
I think this logic is actually needed because `Root / ResourceName("")` yields `s = "/"` and `normalizePrefix` normalizes double slashes only.

BTW both `Root`  and `Root / ResourceName("")` need a prefix of `""`. `"/"` doesn't work

Reply:
Ah right, I confused `//` with `\\`, the latter escaping a single slash.